### PR TITLE
Add support for Find/Install ACR packages with dependencies

### LINKED ISSUE #1585: ACR: Use dependency information for download and installation
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/Microsoft.PowerShell.PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

Install psresource from ACR

### Expected behavior

```console
Dependencies are downloaded
```

### Actual behavior

```console
Dependencies are not downloaded
```

### Error details

_No response_

### Environment data

```powershell
NA
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
This PR adds support specifically for Find and Install to retrieve dependency information for a package. 
Specifically, it parses out dependency information from the server-side metadata Json string and adds that information to the `PSResourceInfo` object.

It also removes some functionality added in PR #1577, namely it removes attaching dependency.json files to a package.  Instead, dependency information is now retrieved from the `RequiredModules` property in the package's metadata.

## PR Context
Resolves #1585 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/ACRServerAPICalls.cs [hunk 1]
```diff
@@ -1136,15 +1122,9 @@ internal bool PushNupkgACR(string psd1OrPs1File, string outputNupkgDir, string p
...
-            TryCreateAndUploadManifest(fullNupkgFile, nupkgDigest, configDigest, pkgName, resourceType, metadataJson, configFilePath,
-                pkgNameLower, pkgVersion, acrAccessToken, out HttpResponseMessage manifestResponse);
-
-            // After manifest is created, see if there are any dependencies that need to be tracked on the server
-            if (dependencies != null && dependencies.Count > 0)
-            {
-                TryProcessDependencies(dependencies, pkgNameLower, acrAccessToken, manifestResponse);
-            }
+            // Create and upload manifest 
+            TryCreateAndUploadManifest(fullNupkgFile, nupkgDigest, configDigest, pkgName, resourceType, metadataJson, configFilePath,
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs:1029]: This method returns a bool which is never used. Should the return of current method be like:

```csharp
return TryCreateAndUploadManifest(fullNupkgFile, nupkgDigest, configDigest, pkgName, resourceType, metadataJson, configFilePath, pkgNameLower, pkgVersion, acrAccessToken);
```

File: src/code/PSResourceInfo.cs [hunk 2]
```diff
@@ -1493,6 +1496,71 @@ public static Dependency[] ParseHttpDependencies(string dependencyString)
             return dependencyList.ToArray();
         }
 
+        public static List<Dependency> ParseACRDependencies(JsonElement requiredModulesElement, out string errorMsg)
```
Comment (adityapatwardhan) [inline src/code/PSResourceInfo.cs]: should this be internal instead?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by anamnavi — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]
Lgtm

**Pull request merged**